internal/pkg/service/stream/storage/level/local/diskreader/reader.go

```diff
@@ -67,8 +69,27 @@ func newReader(
 	ctx = ctxattr.ContextWith(ctx, attribute.String("slice", sliceKey.String()))
 	r.logger.Debug(ctx, "opening disk reader")
 
+	matched, err := filepath.Glob(path)
```

Comment:
The `diskwriter` on the other side writes the files into  `filePath := slice.FileName(volumePath, w.writerKey.SourceNodeID)`. So the path looks on multiple source nodes for example like this:
volume.path = "v1"
slice.FilenamePrefix="slice"
slice.FilenameExtension="gz.csv"
```
tmp/v1/slice-source1.gz.csv
tmp/v1/slice-source2.gz.csv
```
On the reader side we have to read all `.gz` files. Therefore we use `Glob` or any other path finder, that allows us to use `wildcard` symbols to get all the files on all source nodes.

Hopefully I explained it reasonably. If not check this 
https://github.com/keboola/keboola-as-code/blob/main/internal/pkg/service/stream/storage/level/local/diskwriter/writer.go#L80
where the writer writes the file within the volume.

Reply:
Exactly, the previous code was 
```
filePath := slice.FileName(v.Path(), "my-node")
```
Which was statics and had 2 issues.
1. Not configurable source node ID, so the slice was not read unleass `source.NodeID = "my-node"
2. When multiple sources were defined, it read only from 1 instead of `n`.
Basically it would not read whole slice as it would be sliced into multiple files based on `source` node distribution.

Will be tested in E2E test.